File: backend/UMS/Dtos/Shared/BaseResponse.cs
namespace UMS.Dtos.Shared;

public class BaseResponse<T>
{
    public int StatusCode { get; set; } = 500;
    public string? Message { get; set; }
    public T? Result { get; set; }
    public int? Total { get; set; }
    public Pagination? Pagination { get; set; }
}

public class Pagination
{
    public int? CurrentPage { get; set; }
    public int? PageSize { get; set; }
    public int? Total { get; set; }
}

File: mobile/mobile/Models/BaseResponse.swift
import Foundation

struct BaseResponse<T: Codable>: Codable {
    let statusCode: Int
    let message: String
    let result: T?
    let total: Int?
    let pagination: Pagination?
}

struct Pagination: Codable {
    let currentPage: Int?
    let pageSize: Int?
    let total: Int?
}

